src/samVisualize/samVisualizeTypes.ts

```diff
@@ -8,6 +8,8 @@
 export enum LinkTypes {
     GetAtt = 'Fn::GetAtt',
     Sub = 'Fn::Sub',
-    DependsOn = 'DependsOn',
     Ref = 'Ref',
+    DependsOn = 'DependsOn',
+    // A general category to represent any intrinsic function. Includes GetAtt, Sub, and Ref.
+    IntrinsicFunction = 'Intrinsic Function',
 }
```

Comment:
Looks like we have a single enum for both processing and rendering, and either consume all values. I would suggest to use two enums, and having `DependsOn` in both of them is fine

Reply:
Makes sense, I've split them up.